Celeste 1.4.1.0 support

Wartori54 opened the pull request:
Introduces the required changes in Everest to support the publicly available version of Celeste 1.4.1.0. Of course this version may not be final, but all changes have been confirmed by the devs so the fixes are final for sure.

This also forces a downgrade to sdl2, if required. This is the reason the `libfmod_SDL.so` library has to be bundled for linux as well.

Wartori54 committed:
Fix MonoMod patches for 1.4.1.0

maddie480 reviewed (dismissed):
needs updating with the lib-ext rework (though I see a PR is underway to do it already :sweat_smile:)

Wartori54 committed:
Downgrade game to SDL2 if required

Wartori54 commented:
Conflicts solved, bumped lib-ext to include the conflicting change (adding `libfmod_SDL.so` in lib64-linux) in there.

maddie480 committed:
Merge branch 'dev' into 1.4.1.0-support

maddie480 commented:
GitHub Actions and Azure Pipelines can't even agree on which commit is getting built :ok_hand: (GitHub says https://github.com/EverestAPI/Everest/commit/68648f7e6dc9eb823671e86bc2e2f55ac530efc9, Azure says https://github.com/EverestAPI/Everest/commit/aabe20f05a72653fb918b5c084898f41031e341f)

maddie480 commented:
/azp run

maddie480 reviewed (changes requested):
seems like Everest is failing to install? :sweat_smile: 

```
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
  at MonoMod.Cil.ILCursor.GotoNext(MoveType moveType, Func`2[] predicates) in /home/vsts/work/1/s/external/MonoMod/src/MonoMod.Utils/Cil/ILCursor.cs:line 308
  at MonoMod.Cil.ILCursor.GotoNext(Func`2[] predicates) in /home/vsts/work/1/s/external/MonoMod/src/MonoMod.Utils/Cil/ILCursor.cs:line 393
  at MonoMod.MonoModRules.PatchEngineCctor(ILContext context, CustomAttribute attrib)
```

SnipUndercover commented:
Are you installing the version on the SDL3 beta?

maddie480 commented:
no, the install seems to be crashing on 1.4.0.0 (see TAS check pipeline), this needs to be fixed if we don't want to wait for 1.4.1.0 to become default (_if_ this happens) to merge this

Wartori54 committed:
Skip framework print patch on 1.4.0.0

Wartori54 commented:
Fixed. Patching the steam sdl3/1.4.1.0 version on my end works as well.

Note that the .net 8 upgrade also added a print statement to log the exact runtime version, thus on 1.4.1.0 there is some redundancy (both vanilla and Everest show that information). I have not removed it (even tho removing it is trivial) since I find it useful to have all system information in one big block right after the "Booting Everest" log, and the redundancy is not even visible if you are not looking for it specifically.
A very similar redundancy has been present for a long while too with "CELESTE: 1.4.0.0" and the `VersionCelesteString` log.
